src/Jobs/GenerateCSVJob.php

```diff
@@ -223,23 +263,43 @@ protected function outputRows(GridField $gridField, $columns, $start, $count) {
         file_put_contents($this->getOutputPath(), $fileData, FILE_APPEND);
     }
 
-    public function setup() {
+    public function setup()
+    {
         $gridField = $this->getGridField();
         $this->totalSteps = $gridField->getManipulatedList()->count();
     }
 
+    /**
+     * Normally Director::handleRequest will register an HTTPRequest service. If that hasn't happened yet, we will
+     * register one instead. Also register a new controller if one hasn't been pushed yet.
```

Comment:
This comment is good in saying what (i.e. the purpose), but doesn't describe _why_ (I'm assuming that this is necessary in testing, but that's a shot in the dark as is).

Reply:
It can either happen when unit testing, or when running from the command line. When routing via frontend controllers, `Director::handleRequest` registers an HTTPRequest at the start of the method, but this injection isn't consistent throughout core yet, so things like this are sometimes necessary

Reply:
Yeah, sorry I should have been clearer:
I'd like the docblock to include the information you've just put in your reply comment. Perhaps more concisely, but at least an indication for the reader as to why such a method exists far beyond the usual request bootstrap :)

Reply:
Good point, I amended it slightly including the wording from Robbie's comment.